src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,123 @@
+import classNames from 'classnames';
+import { useState } from 'react';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todo: Todo;
+  todosLoadingState: Todo[],
+  onRemoveTodo: (todo: Todo) => void;
+  onToogleUpdateTodo: (todo: Todo) => void;
+  onHandleUpdate: (todo: Todo) => void;
+};
+
+export const TodoItem: React.FC<Props> = ({
+  todo,
+  todosLoadingState,
+  onRemoveTodo,
+  onToogleUpdateTodo,
+  onHandleUpdate,
+}) => {
+  const { title, completed } = todo;
+  const [isEditing, setIsEditing] = useState(false);
+  const [todoTitle, setTodoTitle] = useState(title);
+
+  const eventChange = (e: React.ChangeEvent<HTMLInputElement>) => {
+    const { value } = e.target;
+
+    setTodoTitle(value);
+  };
+
+  const handleOnSubmit = (e: React.FormEvent) => {
+    e.preventDefault();
+    setIsEditing(false);
+
+    const trimTodoTitle = todoTitle.trim();
+
+    if (trimTodoTitle === todo.title) {
+      setTodoTitle(title);
+
+      return;
+    }
```

Comment:
OPTIONAL | you can write it within one line if it doesn't cause any type errros
```suggestion
    if (trimTodoTitle === todo.title) {
      return setTodoTitle(title);
    }
```